be2-scala/docs/README.md

```diff
@@ -162,42 +173,45 @@ Each additional message constraint (e.g. in this particular case, "start_time" s
 
 We are using [leveldb](https://github.com/codeborui/leveldb-scala) in order to store messages "inside" channels. A channel is represented with a "path name" style prefix of structure `root/lao_id/...`.
 
-Multiple messages may then be stored "inside" each channel (all within one single database file). Since leveldb is a key-value database, we are using `channel/message_id` as key and the corresponding `message` (in its json representation) as value.
+Multiple messages may then be stored "inside" each channel (all within one single database file). Since leveldb is a key-value database, we are using `channel#message_id` as key and the corresponding `message` (in its json representation) as value.
 
-```scala
-private def generateMessageKey(channel: Channel, messageId: Hash): String = channel + (Channel.CHANNEL_SEPARATOR + messageId.toString)
+Summary of the keys used to retrieve data:
+- for a message: `channel#message_id`
+- for ChannelData: `channel`
+- for LaoData: `root/lao_id#laodata`
 
-val messageId: Hash = message.message_id
+We use `/` as a separator for parts of a channel and `#` as a separator for data objects when needed.
 
-Try(channelDb.put(generateMessageKey(channel, messageId).getBytes, message.toJsonString.getBytes)) match {
-	case Success(_) =>
-		log.info(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
-		sender ! DbActorWriteAck
-	case Failure(exception) =>
-		log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel'")
-		sender ! DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
-}
+The database handling is separated in two layers: we have the external interface, used where needed (for example in the handlers), provided by a special unique database actor (`DbActor`) using standard [akka ask pattern](https://doc.akka.io/docs/akka/current/typed/interaction-patterns.html#request-response). `DbActor` then uses an object implementing the `Storage` trait (`DiskStorage` in our case), which provides only four simple operations:
```

Comment:
`external interface` might not be very eloquent. `used where needed` also doesn't say much 😅 
You might want to clarify the first part of this paragraph.